Build issues

Trying to build jim on Win7 x86 with TDM-GCC10.3 and having Magicsplat TCL installed.
Getting this:
$ make                                                                                                                          
        MKLDEXT _load-static-exts.c                                                                                             
        CC      _load-static-exts.o                                                                                             
        CC      jim-subcmd.o                                                                                                    
        CC      jim-interactive.o                                                                                               
        CC      jim-format.o                                                                                                    
        CC      jim.o                                                                                                           
jim.c: In function 'JimClock':                                                                                                  
jim.c:701:5: warning: 'gettimeofday' is deprecated [-Wdeprecated-declarations]                                                  
  701 |     gettimeofday(&tv, NULL);                                                                                            
      |     ^~~~~~~~~~~~                                                                                                        
In file included from jim.c:65:                                                                                                 
C:/TDM-GCC-32/include/sys/time.h:106:53: note: declared here                                                                    
  106 | int __cdecl __MINGW_NOTHROW __POSIX_2008_DEPRECATED gettimeofday                                                        
      |                                                     ^~~~~~~~~~~~                                                        
        CC      utf8.o                                                                                                          
        CC      jimregexp.o                                                                                                     
        CC      jimiocompat.o                                                                                                   
jimiocompat.c: In function 'JimProcessPid':                                                                                     
jimiocompat.c:98:12: warning: implicit declaration of function 'GetProcessId'; did you mean 'GetProcessHeap'? [-Wimplicit-functi
on-declaration]                                                                                                                 
   98 |     return GetProcessId(pid);                                                                                           
      |            ^~~~~~~~~~~~                                                                                                 
      |            GetProcessHeap                                                                                               
        CC      linenoise.o                                                                                                     
        CC      jim-win32compat.o                                                                                               
        CC      jim-nosignal.o                                                                                                  
        CC      jim-aio.o                                                                                                       
jim-aio.c: In function 'aio_cmd_stat':                                                                                          
jim-aio.c:1433:16: error: storage size of 'sb' isn't known                                                                      
 1433 |     jim_stat_t sb;                                                                                                      
      |                ^~                                                                                                       
jim-aio.c:1433:16: warning: unused variable 'sb' [-Wunused-variable]                                                            
jim-aio.c:1441:1: warning: control reaches end of non-void function [-Wreturn-type]                                             
 1441 | }                                                                                                                       
      | ^                                                                                                                       
make: *** [jim-aio.o] Error 1

--- a/auto.def
+++ b/auto.def
@@ -285,6 +285,8 @@ switch -glob -- $host_os {
         define TCL_PLATFORM_OS $host_os
         define TCL_PLATFORM_PLATFORM windows
         define TCL_PLATFORM_PATH_SEPARATOR {;}
+        # Target WinXP or later. Should this be configurable?
+        define-append CCOPTS -D_WIN32_WINNT=0x501 -Wno-deprecated-declarations
     }
     default {
         # Note that cygwin is considered a unix platform

--- a/jimiocompat.h
+++ b/jimiocompat.h
@@ -65,8 +65,8 @@ int Jim_OpenForRead(const char *filename);
     #define HAVE_PIPE
     #define pipe(P) _pipe((P), 0, O_NOINHERIT)
 
-    typedef struct _stat64 jim_stat_t;
-    #define Jim_Stat __stat64
+    typedef struct __stat64 jim_stat_t;
+    #define Jim_Stat _stat64
     #define Jim_FileStat _fstat64
 
 #else

